.github/workflows/on-code-change.yml

```diff
@@ -0,0 +1,31 @@
+name: On Code Change
+
+on: [push]
+
+jobs:
+  lint-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        shell: bash
+        run: yarn install --frozen-lockfile
+      - name: Run linting
+        shell: bash
+        run: yarn lint
+      - name: Run tests
+        shell: bash
+        run: yarn test
+    
+  build:
+    runs-on: ubuntu-latest
+    needs: lint-tests
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        shell: bash
+        run: yarn install --frozen-lockfile
+      - name: Run Build
+        shell: bash
+        run: yarn build
```

Comment:
The build process must be only triggered when we push on the `master` branch.
Can you split it to an another workflow for the build process ?